packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -242,47 +287,81 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
   const [inputValue, setInputValue] = React.useState<string>(
     inputProps?.value?.toString() ?? '',
   );
+
   const handleOnInput = React.useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
       const { value } = event.currentTarget;
       setInputValue(value);
-      dispatch(['open']); // reopen when typing
+      dispatch({ type: 'open' }); // reopen when typing
       setFilteredOptions(
         filterFunction?.(optionsRef.current, value) ??
           optionsRef.current.filter((option) =>
             option.label.toLowerCase().includes(value.toLowerCase()),
           ),
       );
       if (focusedIndex != -1) {
-        dispatch(['focus', -1]);
+        dispatch({ type: 'focus', value: -1 });
       }
       inputProps?.onChange?.(event);
     },
     [filterFunction, focusedIndex, inputProps, optionsRef],
   );
 
-  // When the value prop changes, update the selectedIndex
+  // When the value prop changes, update the selected
```

Comment:
This would be a bit more clear if you changed the comment to "When the value prop changes, update selected" or "When the value prop changes, update the selected index/indices"

Reply:
Updated comment